boards/arm/weact_stm32g431_core/weact_stm32g431_core.dts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2023 Andreas Sandberg
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/g4/stm32g431Xb.dtsi>
+#include <st/g4/stm32g431c(6-8-b)ux-pinctrl.dtsi>
+#include <zephyr/dt-bindings/usb-c/pd.h>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	model = "WeAct Studio STM32G431 Core Board";
+	compatible = "weact,stm32g431-core";
+
+	chosen {
+		zephyr,console = &usart1;
+		zephyr,shell-uart = &usart1;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+	};
+
+	aliases {
+		led0 = &led_0;
+		mcuboot-button0 = &button_0;
+		mcuboot-led0 = &led_0;
+		sw0 = &button_0;
+		sw1 = &button_1;
+		usbc-port0 = &usbc1;
+		watchdog0 = &iwdg;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led_0: led0 {
+			gpios = <&gpioc 6 GPIO_ACTIVE_HIGH>;
+			label = "Status LED";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		button_0: button0 {
+			label = "User";
+			gpios = <&gpioc 13 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
+			zephyr,code = <INPUT_KEY_0>;
+		};
+		button_1: button1 {
+			label = "Boot0";
+			gpios = <&gpiob 8 GPIO_ACTIVE_HIGH>;
+			zephyr,code = <INPUT_KEY_1>;
+		};
+	};
+
+	vbus1: vbus {
+		compatible = "zephyr,usb-c-vbus-adc";
+		status = "disabled";
+		io-channels = <&adc2 12>;
+		output-ohms = <10000>;
+		full-ohms = <(100000 + 10000)>;
+	};
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		usbc1: usbc-port@1 {
+			compatible = "usb-c-connector";
+			status = "disabled";
+			reg = <1>;
+			tcpc = <&ucpd1>;
+			vbus = <&vbus1>;
+			data-role = "device";
+			power-role = "sink";
+			sink-pdos = <PDO_FIXED(5000, 100, PDO_FIXED_USB_COMM)>;
+		};
+	};
+};
+
+&clk_lsi {
+	status = "okay";
+};
+
+&clk_lse {
+	status = "okay";
+};
+
+&clk_hsi {
+	status = "okay";
+};
```

Comment:
Do you actually need clk_hsi and clk_lsi ?

Reply:
My understanding is that the IWDG always uses the LSI and the UCPD1 block always uses the HSI. I'll disable the HSI by default and enable it in the USB-C sample since the UCPD1 block is disabled here.